lib/Remotery.c

```diff
@@ -4233,11 +4233,11 @@ static void Remotery_Destructor(Remotery* rmt)
     // Join the remotery thread before clearing the global object as the thread is profiling itself
     Delete(Thread, rmt->thread);
 
-    // Ensure this is the module that created it
-    assert(g_RemoteryCreated == RMT_TRUE);
-    assert(g_Remotery == rmt);
-    g_Remotery = NULL;
-    g_RemoteryCreated = RMT_FALSE;
+    if (g_RemoteryCreated)
+    {
+      g_Remotery = NULL;
+      g_RemoteryCreated = RMT_FALSE;
+    }
 
```

Comment:
Right. The "assign to global pointer" pattern is broken in there anyway and doesn't really adhere to the New/Delete lifetime pattern (`Remotery_ThreadMain` needs to be created outside the Remotery constructor). But there is value in bringing it closer.